src/Functions/if.cpp

```diff
@@ -1116,6 +1116,29 @@ class FunctionIf : public FunctionIfBase
 
         return res;
     }
+
+    ColumnPtr getConstantResultForNonConstArguments(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type) const override
+    {
+        const ColumnWithTypeAndName & arg_cond = arguments[0];
+        if (!arg_cond.column || !isColumnConst(*arg_cond.column))
+            return {};
+
+        const ColumnConst * cond_const_col = checkAndGetColumnConst<ColumnVector<UInt8>>(arg_cond.column.get());
+        bool condition_value = cond_const_col->getValue<UInt8>();
```

Comment:
Here is nullptr dereference, apparently the cond is not always `ColumnConst<ColumnVector<UInt8>>`